plat/hikey/plat_pm.c

```diff
@@ -222,9 +225,38 @@ static void __dead2 hikey_system_off(void)
 	 * through Jumper 1-2. So, to complete shutdown, user needs to manually
 	 * remove Jumper 1-2.
 	 */
-	NOTICE(".........................................................\n");
-	NOTICE(" IMPORTANT: Remove Jumper 1-2 to complete shutdown\n");
-	NOTICE(" DANGER:    SoC is still burning. DANGER!\n");
+	/* init timer00 */
+	mmio_write_32(TIMER00_CONTROL, 0);
+	mmio_write_32(TIMER00_LOAD, 0xffffffff);
+	/* free running */
+	mmio_write_32(TIMER00_CONTROL, 0x82);
+
+	/* adding delays */
+	start = mmio_read_32(TIMER00_VALUE);
+	do {
+		cnt = mmio_read_32(TIMER00_VALUE);
+		if (cnt > start) {
+			delta = 0xffffffff - cnt;
+			delta += start;
+		} else
+			delta = start - cnt;
+		delta_ms = delta / 19200;
+		if (delta_ms > 1000 && show) { /* after 1 second */
+			/* if we are still alive, that means Jumper
+			 * 1-2 is mounted. Need to warn and reboot
+			 */
+			NOTICE("..........................................\n");
+			NOTICE(" IMPORTANT: Remove Jumper 1-2 to shutdown\n");
+			NOTICE(" DANGER:    SoC is still burning. DANGER!\n");
```

Comment:
So, I think its good to display info as to why the board reboots instead of powers-down here, but I think we can get rid of the DANGER/SoC burning wording. Its not really dangerous anymore because the board is about to reboot.